### PR TITLE
Bootloader rollback protection

### PR DESCRIPTION
**Pull request by mike-sul**
- Tests&Fixtures refactoring.
- Reject update if bootloader rollback is detected and rollback protection is turned on.
- Add tests for the rollback protection.

### CODE REVIEW

**Comment by mike-sul** [comment]
@igoropaniuk You may wanna look into this https://github.com/foundriesio/aktualizr-lite/pull/232/commits/84ecb919131325a09ae1d5ab3c313bfafdecb61e. Please, let me know if it's what you have expected?

**Comment by igoropaniuk** [comment]
@mike-sul https://github.com/foundriesio/aktualizr-lite/commit/84ecb919131325a09ae1d5ab3c313bfafdecb61e LGTM, thanks!

File: src/bootloader/bootloaderlite.h [hunk 1]
```diff
@@ -25,6 +25,9 @@ class BootFwUpdateStatus {
 
 class BootloaderLite : public Bootloader, public BootFwUpdateStatus {
  public:
+  using VersionType = uint64_t;  // clang-tidy prefers uint64_t over unsigned long long int
+  using VersionNumbRes = std::tuple<VersionType, bool>;
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply src/bootloader/bootloaderlite.h:29]: It's used as a return type of `getDeploymentVersion` and `getCurrentVersion` methods.

File: tests/fixtures/liteclient/boot_flag_mgr.cc [hunk 2]
```diff
@@ -38,6 +38,9 @@ class BootFlagMgr {
   void set_bootupgrade_available() {
     Utils::writeFile(dir_/"bootupgrade_available", std::string("1"));
   }
+  void set_bootfirmware_version(const std::string& ver) {
+    Utils::writeFile(dir_/"bootfirmware_version", ver);
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply tests/fixtures/liteclient/boot_flag_mgr.cc:45]: Yep, that's true. I just think this is a matter of another task/ticket to fix/improve it.

File: tests/ostree_test.cc [hunk 3]
```diff
@@ -4,26 +4,17 @@
...
-
+  OSTreeTest():path_{test_dir_.PathString()}, repo_{std::make_unique<OSTree::Repo>(path_, true)} {}
   bool isRepoInited() const {
     return boost::filesystem::exists(path_ + "/config") && boost::filesystem::exists(path_ + "/objects");
   }
 
  protected:
-  std::string path_;
+  TemporaryDirectory test_dir_;  // must be the first element in the class
+  const std::string path_;
```
Comment (doanac) [inline tests/ostree_test.cc:22]: hey - C++ doesn't *always* have to suck :)

**Comment by mike-sul** [comment]
@doanac Did you have a chance to look into it?

**Review by doanac — APPROVED** [review]
rollback is now beyond what i can understand. The code seems to add up correctly, so if its passing the tests - i'm happy.

**Pull request merged**